### DISCUSSION TITLE
Import to exclude/include custom list from csv file

### DISCUSSION THREAD

**Discussion by waynecarnold** [mcguirepr89/BirdNET-Pi#983] [Ideas]
Title says it all. Currently it is very time consuming to add to the lists.

**Comment by willkuhn** [comment]
I've got an easy solution for you 😄 

**To create or edit a custom exclude list:**

1. Go to _Tools_ > _File Manager_
2. In the File Manager, navigate to `[your_root_folder_name] / BirdNET-Pi / exclude_species_list.txt`
3. Click _Advanced Editor_
4. Use the editor to add in or edit your list of species to exclude, one species per line (see notes below on format)
5. Click _Save_
6. Your updates should immediately appear under the "Excluded Species List" column in _Tools_ > _Excluded Species List_. You can further tweak the list in this viewer or in the Advanced Editor as above

Follow the same process **to create/edit a custom include list**, stored under `[your_root_folder_name] / BirdNET-Pi / include_species_list.txt`. But heed this **warning** from BirdNET-Pi: "If this list contains ANY species, the system will ONLY recognize those species. Keep this list EMPTY unless you are ONLY interested in detecting specific species."

**Notes:**

- BirdNET-Pi will only recognize species names in a specific format (e.g., "Accipiter cooperii_Cooper's Hawk"). These correspond to names of bird species, non-bird species and other sounds that the BirdNET network is trained to detect, i.e., `labels`
- Get a full list of the ~6500K possible `labels` your version of BirdNET can detect in the File Manager at `[your_root_folder_name] / BirdNET-Pi / model / labels.txt`. I downloaded the text file, copy-pasted into excel to tweak there, then copy-pasted my final exclude list into the `exclude_species_list.txt` via the text editor, which seemed to work well
- As far as I can tell, any values on the exclude/include lists that DO NOT correspond to one of these BirdNET labels will just be ignored and do not appear to cause any problems. This includes any line that contains multiple species names (should be just one name per line). You can see and remove any unrecognized values in the viewer. That being said, it's a good idea to backup any custom lists before messing with them as describe here
- You could use the same process to copy-paste values from a CSV file. You can probably also just overwrite `exclude_species_list.txt` or `include_species_list.txt` in the File Manager or terminal, if done with care and caution

I hope this helps!
